PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -3130,7 +3202,57 @@ function Get-SddcDiagnosticInfo
     # Phase 4
     #
 
-    Show-Update "<<< Phase 4 - Compacting files for transport >>>" -ForegroundColor Cyan
+    if ($ZipFiles)
+    {
+        Show-Update "<<< Phase 4 - Compacting files for transport >>>" -ForegroundColor Cyan
+    }
+    else
+    {
+        Show-Update "<<< Phase 4 - Extract cab files + Final Cleanup >>>" -ForegroundColor Cyan
+    }
+
+    if (!$ZipFiles)
+    {
+         Show-update "Extract cab files"
+         $items = get-childitem -Recurse -Path $Path -Filter "*.cab"
+         foreach ($item in $items)
+         {
+	        # make a new directory for .cab contents
+	        $parentPath = Split-Path -Path $item.fullname -Parent
+	        $newDir = Join-Path -Path $parentpath -ChildPath $item.basename
+	        mkdir $newdir
+
+            try
+            {
+                cmd.exe /c C:\Windows\System32\expand.exe -F:* $item.FullName $newdir
+            }
+            catch
+            {
+                Write-Warning "Extracting .cab contents failed with exception : $_"
+            }
+         }
+         $items | Remove-Item -recurse -force
+
+         Show-Update "Rename msdbg.<node name> to msdbg. Do this to shorten overall filepath."

```

Comment:
Good as far as it goes, but I don't know that 15 characters is enough?

Reply:
This has been working for us.